### ISSUE TITLE
Handle Macros when Navigating to Defintion

### ISSUE THREAD

**Issue by kbuzby** [microsoft/vscode-cpptools#1032]
It would be nice if the "Go To Definition (F12)" option when navigating C/C++ files could handle macros that are defined in the normal include paths.

Version info:
OS: Lubuntu 16.04
VSCode: 1.17.0-Insider
cpptools: 0.12.4

**Comment by sean-mcmanus** [comment]
What do you mean by "normal include paths"? Go to Definition works with macros and is currently using the Tag Parser, so the definitions are found recursively from the `browse.path` setting (not `includePath`).

**Comment by kbuzby** [comment]
Couple things I messed up. I did mean browse.path when referring to "include paths"

The project I'm working on has a separate include directory for some header files, adding that directory specifically to the browse.path (and includePath to get rid of warning) did fix the issue.

However, the project has many *.d files created as well, which would have pointed to the proper header files. Could these be used to also search for definitions?
